syntax = "proto3";

package jobpostings;

// Represents a job posting.
message Job {
  string id = 1;
  string title = 2;
  string company_name = 3;
  string description = 4;
  string location = 5;
  int32 views = 6; // Defaults to 0 if not provided.
}


// -----------------------------
// Average Salary Messages
// -----------------------------
message AverageSalaryRequest {
  string title = 1;
}

message AverageSalaryResponse {
  float averageSalary = 1;
}

// -----------------------------
// Job Posting Messages
// -----------------------------

message JobAddRequest {
    string title = 1;
    float normalized_salary =2;
    string company_name = 3;
    string description = 4;
    string location = 5;
}

message JobAddResponse {
    string message = 1;
    int32 status = 2;
}

// -----------------------------
// Service Definition
// -----------------------------

// Service handling job posting functionalities.
service JobPostingService {
  rpc AverageSalary(AverageSalaryRequest) returns (AverageSalaryResponse);
  rpc AddJob(JobAddRequest) returns (JobAddResponse);  
}
